### ISSUE TITLE
ARG Variables are substituted with empty string when using cache

### ISSUE THREAD

**Issue by BenHizak** [GoogleContainerTools/kaniko#764]
**Actual behavior**
Kaniko substitutes ARG with `(empty string)` when using cache. When not using cache, the behavior is normal.

**Expected behavior**
Kaniko should substitute arguments.

**To Reproduce**
See reproducing steps here: https://gist.github.com/BenHizak/4632b3df40892a46712f29b4cf4a6057

Using the following Kaniko image
`gcr.io/kaniko-project/executor@sha256:9000801241318d6c9b8c2b505974f22fd74c57e0adfdb09c98870aef4923bb65`

**Comment by donmccasland** [comment]
Probably just log output issue.  Need to make sure we filter credential based ENV vars.

**Comment by tejal29** [comment]
Unfortunately, we don't evaluate the `ARG` and hence the arg substitution is empty.

**Comment by cvgw** [comment]
@tejal29 I'm assuming we will never log the value of ARG because they may contain sensitive data like SSH keys. WDYT?

**Comment by BenHizak** [comment]
Why not let users decide? it makes it difficult to see important elements
of the CI pipeline. Is there a way to denote specific ARGs to show?

**Comment by tejal29** [comment]
> @tejal29 I'm assuming we will never log the value of ARG because they may contain sensitive data like SSH keys. WDYT?

No, we never added the ARG value to `confif.ENV` which ends up replacing all the places which use the arg. 

I added a PR to fix this. 

You can now see the arg in log
```
 # /kaniko/executor -f dockerfiles/Dockerfile1 --context=dir://workspace --destination=gcr.io/tejal-test/test --tarPath=image.tar
INFO[0000] Resolved base name python:${VERSION} to python: 
INFO[0000] Using dockerignore file: /workspace/.dockerignore 
INFO[0000] Resolved base name python:${VERSION} to python: 
INFO[0000] Retrieving image manifest python:3.7.2       
INFO[0001] Retrieving image manifest python:3.7.2       
INFO[0002] Built cross stage deps: map[]                
INFO[0002] Retrieving image manifest python:3.7.2       
INFO[0002] Retrieving image manifest python:3.7.2   
```

**Comment by tejal29** [comment]
> @tejal29 I'm assuming we will never log the value of ARG because they may contain sensitive data like SSH keys. WDYT?

No. This only happens when parsing stages from dockerfile. At the time of parsing stages, we only evaluate args present in [`--build-args`](https://github.com/GoogleContainerTools/kaniko/blob/649a0ed99ba704fe3d9c18c77de2bbd1bdec61ed/pkg/dockerfile/dockerfile.go#L67). 

None of the dockerfile commands have been evaluated at this time and hence `ARG VERSION=3.7` is not evaluated

**Comment by cvgw** [comment]
> Why not let users decide? it makes it difficult to see important elements of the CI pipeline. Is there a way to denote specific ARGs to show?

Would adding an additional run command to your dockerfile be a work around?

I.E.

```
....
RUN echo $ARG_I_WANT_TO_SEE_LOGGED
....
```

**Comment by wanderrful** [comment]
> Probably just log output issue. Need to make sure we filter credential based ENV vars.

If we filter credential-based ENV vars from being able to get printed normally, how can we confirm that they are being injected properly into the build?
